### PR TITLE
Missing @ProviderType

### PR DESCRIPTION
**Pull request by hongvo2308**

### CODE REVIEW

File: develop/tutorials/articles/120-service-builder/02-service-builder-persistence/05-developing-custom-sql-queries.markdown [hunk 1]
```diff
@@ -107,6 +107,7 @@ Run Service Builder to generate the `*Finder` interface, which is based on
 the `*FinderImpl` class. Modify your `*FinderImpl` class to have it implement
 the `*Finder` interface you just generated:
```
Comment (VyBui) [inline develop/tutorials/articles/120-service-builder/02-service-builder-persistence/05-developing-custom-sql-queries.markdown]: Keep up the good work Hong

**Comment by jhinkey** [comment]
@hongvo2308 - I don't see how `@ProviderType` applies to this implementation class--it's not an interface. Can you explain why you think it's needed? 

Thanks,
Jim

**Comment by hongvo2308** [comment]
@jhinkey Yes, In my case, without @ProviderType, my Finder can not run but with the @ProviderType, it can run. So I have thought that this annotation may be the problem. And I see other class *PersistenceImpl also have it. So I have assumed that this annotation is missing.

**Comment by jhinkey** [comment]
@rotty3000 is this example implementation a provider type?

@hongvo2308 What errors do you get when you try to install and execute this finder?

**Comment by rotty3000** [comment]
@jhinkey I think just to be consistent with Liferay code. The it's applied here in Liferay code is because FinderImpls are used for further code generation, which we want to apply @ProviderType. So it's a really narrow corner case. To be honest, I would consider it confusing for the example.

**Comment by hongvo2308** [comment]
@jhinkey Yes, I have tested again without this annotation and the function works well. It seems that I make another mistake at the first time. So the annotation is not the problem.
I will close this pr.
Thank you for your time.

**Comment by jhinkey** [comment]
No problem @hongvo2308 . Glad it's working for you!

**Pull request closed without merging**